fdbserver/IPager.h

```diff
@@ -76,16 +77,69 @@ static const std::vector<std::pair<PagerEvents, PagerEventReasons>> L0PossibleEv
 	{ PagerEvents::PageWrite, PagerEventReasons::MetaData },
 };
 
-// Represents a block of memory in a 4096-byte aligned location held by an Arena.
+enum EncodingType : uint8_t {
+	XXHash64 = 0,
+	// For testing purposes
+	XOREncryption = 1
+};
+
+enum PageType : uint8_t {
+	HeaderPage = 0,
+	BackupHeaderPage = 1,
+	BTreeNode = 2,
+	BTreeSuperNode = 3,
+	QueuePageStandalone = 4,
+	QueuePageInExtent = 5
+};
+
+// Encryption key ID
+typedef uint64_t KeyID;
+
+// Encryption key and secret string
+struct KeyIDWithSecret {
+	KeyID id;
+	Key secret;
+};
+
+// Interface used by pager to get encryption keys by ID when reading pages from disk
+// and by the BTree to get encryption keys to use for new pages
+class IEncryptionKeyProvider {
+public:
+	virtual Future<Key> getSecretByKeyID(KeyID keyID) = 0;
+	virtual Future<KeyIDWithSecret> selectKey(const KeyRef begin, const KeyRef end) = 0;
```

Comment:
For better understanding: 
"selectKey" is equivalent to "getKeyForKeyRange" ? If yes, can we rename it explicitly. For encrypt role and for GS endpoints, plan is to introduce two new endponts:
1. getKeyByID
2. getKeyByKeyString (encrypt role would perform necessary transformation from range -> keyString as GS doesn't understand ranges I think).
Thoughts?

Reply:
Hah, yeah I already renamed it before I read this comment.